rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -645,8 +654,29 @@ fn build_predicate_expression(
     use crate::logical_plan;
     // predicate expression can only be a binary expression
     let (left, op, right) = match expr {
-        Expr::BinaryExpr { left, op, right } => (left, *op, right),
+        Expr::BinaryExpr { left, op, right } => (left.clone(), *op, right.clone()),
+        Expr::Between {
+            expr,
+            negated,
+            low,
+            high,
+        } if !negated // NOT BETWEEN not supported
+            => {
+            let left = Box::new(Expr::BinaryExpr {
+                left: expr.clone(),
+                op: Operator::GtEq,
+                right: low.clone(),
+            });
+            let right = Box::new(Expr::BinaryExpr {
+                left: expr.clone(),
+                op: Operator::LtEq,
+                right: high.clone(),
+            });
+
+            (left, Operator::And, right)
+        }
         _ => {
+            debug!("Filter expression not supported in ParquetExec {:?}", expr);
```

Comment:
thank you for adding some `debug!` logs, I have been thinking how predicate push-down into parquet could be displayed into `explain` output for easier diagnostics, but lately I have been preoccupied with rethinking of reading of arrow arrays from parquet files

Reply:
Yeah this is very useful! I found that the filter pushdown doesn't filter out *anything* on the TPC-H benchmark (even when adding some extra support), as the data is distributed so evenly, so every row group contains almost the complete range of values.

I think @andygrove was thinking about some more general support for keeping statistics like this, but some debug logging is something easy to add.

Reply:
Very interested in your progress with Parquet to Arrow! I was hoping this PR would do something for the tpch benchmark, but seems the filter pushdown is not very useful in this case :)

Reply:
yes, filter push-down into parquet works best when the filter is over sorted data, if the data is randomly distributed across files it can't do much for performance as you have found

Reply:
@Dandandan regarding progress on reading arrow arrays from parquet files, I have been able to make some progress and hope to be able to update the jira issue with my latest findings in the next day or two